terminus-ui/date-range/testing/src/test-helpers.ts

```diff
@@ -7,10 +7,77 @@ import {
 } from '@angular/forms';
 import { DebugElement } from '@angular/core';
 import { TsInputComponent } from '@terminus/ui/input';
+import { TsDateRangeComponent } from '@terminus/ui/date-range';
 
 
+/**
+ * Get an array of all TsDateRangeComponent debug elements
+ *
+ * @param fixture - The component fixture
+ * @return The array of DebugElements
+ */
+export function getAllDateRangeDebugElements(fixture: ComponentFixture<any>): DebugElement[] {
+  const debugElements = fixture.debugElement.queryAll(By.css('ts-date-range'));
+  if (!debugElements || debugElements.length < 1) {
+    throw new Error(`'getAllDateRangeInstances' found no date range instances`);
+  }
+  return debugElements;
+}
+
+/**
+ * Get all TsDateRangeComponent instances
+ *
+ * @param fixture - The component fixture
+ * @return The array of TsCheckboxComponent instances
+ */
+export function getAllDateRangeInstances(fixture: ComponentFixture<any>): TsDateRangeComponent[] {
+  return getAllDateRangeDebugElements(fixture).map((i) => i.componentInstance);
+}
+
+/**
+ * Get an array of debug elements for TsInputComponents within a single TsDateRangeComponent
+ *
+ * @param fixture - The component fixture
+ * @param index - The index of the TsDateRangeComponent
+ * @return The array of DebugElements
+ */
+export function getDateRangeInputDebugElements(fixture: ComponentFixture<any>, index = 0): DebugElement[] {
+  const debugElement = getAllDateRangeDebugElements(fixture)[index];
+  if (!debugElement) {
+    throw new Error(`'getDebugRangeInputs' found no debug element at '${index}'`);
+  }
+  return debugElement.queryAll(By.css('ts-input'));
+}
+
+/**
+ * Get an array of TsInputComponent instances for a single TsDateRangeComponent
+ *
+ * @param fixture - The component fixture
+ * @param index - The index of the TsDateRangeComponent
+ * @return The array of TsInputComponent instances
+ */
+export function getRangeInputInstances(fixture: ComponentFixture<any>, index = 0): TsInputComponent[] {
+  return getDateRangeInputDebugElements(fixture, index).map((v) => v.componentInstance);
+}
 
+/**
+ * Get an array of HTMLInputElements for a single TsDateRangeComponent
+ *
+ * @param fixture - The component fixture
+ * @param index - The index of the TsDateRangeComponent
+ * @return The array of TsInputComponent instances
+ */
+export function getRangeInputElements(fixture: ComponentFixture<any>, index = 0): HTMLInputElement[] {
+  return getDateRangeInputDebugElements(fixture, index).map((v) => v.componentInstance.inputElement.nativeElement);
+}
```

Comment:
I wonder what's the use case here... for a single TsDateRangeComponent, there are two TsInputComponents. When an index passed in, there's just one input returned, why an array is needed here?

Reply:
In this case, the index is used for getting a specific `TsDateRangeComponent`. This is to support consumers that may have more than one date range in a view. Once we get the specific date range, then we return an array of both inputs within that component.